scripts/candlesticks_example.py

```diff
@@ -0,0 +1,99 @@
+from typing import Dict
+
+import numpy as np
+import pandas as pd
+
+from hummingbot.connector.connector_base import ConnectorBase
+from hummingbot.data_feed.candles_feed.candles_factory import CandlesFactory
+from hummingbot.strategy.script_strategy_base import ScriptStrategyBase
+
+
+class SignalFactory:
+    def __init__(self, max_records: int, connectors: Dict[str, ConnectorBase], interval: str = "1m"):
+        self.connectors = connectors
+        self.candles = {
+            connector_name: {trading_pair: CandlesFactory.get_candle(connector="binance_perpetuals",
+                                                                     trading_pair=trading_pair,
+                                                                     interval=interval, max_records=max_records)
+                             for trading_pair in connector.trading_pairs} for
+            connector_name, connector in self.connectors.items()}
+
+    def stop(self):
+        for connector_name, trading_pairs_candles in self.candles.items():
+            for candles in trading_pairs_candles.values():
+                candles.stop()
+
+    def start(self):
+        for connector_name, trading_pairs_candles in self.candles.items():
+            for candles in trading_pairs_candles.values():
+                candles.start()
+
+    @property
+    def all_data_sources_ready(self):
+        return all(np.array([[candles.is_ready for trading_pair, candles in trading_pairs_candles.items()]
+                             for connector_name, trading_pairs_candles in self.candles.items()]).flatten())
+
+    def candles_df(self):
+        return {connector_name: {trading_pair: candles.candles for trading_pair, candles in
+                trading_pairs_candles.items()}
+                for connector_name, trading_pairs_candles in self.candles.items()}
+
+    def features_df(self):
+        candles_df = self.candles_df().copy()
+        for connector_name, trading_pairs_candles in candles_df.items():
+            for trading_pair, candles in trading_pairs_candles.items():
+                candles.ta.rsi(length=14, append=True)
+        return candles_df
+
+    def current_features(self):
+        return {connector_name: {trading_pair: features.iloc[-1, :].to_dict() for trading_pair, features in
+                                 trading_pairs_features.items()}
+                for connector_name, trading_pairs_features in self.features_df().items()}
+
+
+class CandlesticksExample(ScriptStrategyBase):
+    max_executors_by_connector_trading_pair = 1
+    trading_pairs = ["ETH-USDT"]
+    exchange = "binance_paper_trade"
+    markets = {exchange: set(trading_pairs)}
+
+    def __init__(self, connectors: Dict[str, ConnectorBase]):
+        super().__init__(connectors)
+        self.signal_factory = None
+
+    def on_tick(self):
+        if not self.signal_factory:
+            self.signal_factory = SignalFactory(max_records=2500, connectors=self.connectors, interval="1d")
```

Comment:
Make the default values more reasonable, so that it doesn't throw an error, maybe something like
`max_records=200, connectors=self.connectors, interval="1m")`

Also, add comments that help the user know what options to set here:
- possible values for interval
- what does max records do?